### PR TITLE
Add Tabby Docker Space docs

### PR DESCRIPTION
**Pull request by wsxiaoys**

### CODE REVIEW

**Review by julien-c — COMMENTED** [review]
looks cool! 

do you have an example of a running instance on a Space?

**Comment by wsxiaoys** [comment]
I have just unpaused the template space that was being used in the document: 

Link: https://huggingface.co/spaces/TabbyML/tabby-template-space
Direct URL: https://tabbyml-tabby-template-space.hf.space/

**Comment by julien-c** [comment]
pinging @mishig25 for a review as he's very interested in code generation extensions!

File: docs/hub/spaces-sdks-docker-tabby.md [hunk 1]
```diff
@@ -0,0 +1,60 @@
...
+
+[![Deploy on HF Spaces](https://huggingface.co/datasets/huggingface/badges/raw/main/deploy-to-spaces-lg.svg)](https://huggingface.co/spaces/TabbyML/tabby-template-space?duplicate=true)
+
+You need to define the Owner (your personal account or an organization), a Space name, and the Visibility. To secure the api endpoint, we're configuring the visibility as Private.
+
+![image](https://github.com/huggingface/hub-docs/assets/388154/b00eb7ea-7de6-4832-b3ff-512859bfa92e)
+
+<Tip>
+If you want to customize the title, emojis, and colors of your space, go to "Files and Versions" and edit the metadata of your README.md file.
+</Tip>
```
Comment (mishig25) [inline docs/hub/spaces-sdks-docker-tabby.md:23]:
```suggestion
<Tip>

If you want to customize the title, emojis, and colors of your space, go to "Files and Versions" and edit the metadata of your README.md file.

</Tip>
```

we need newlines here

File: docs/hub/spaces-sdks-docker-tabby.md [hunk 2]
```diff
@@ -0,0 +1,60 @@
...
+
+You’ll see the Building status and once it becomes Running your space is ready to go. If you don’t see the Tabby swagger UI, try refreshing the page.
+
+![image](https://github.com/huggingface/hub-docs/assets/388154/f2320e55-7aad-42ed-8b1c-aec4e42e05f0)
+
+### Your Tabby Space URL
+Once Tabby is running, you can use the UI with the <u>Direct URL</u> in the **Embed this Space** option (top right).
```
Comment (mishig25) [inline docs/hub/spaces-sdks-docker-tabby.md:30]:
```suggestion
### Your Tabby Space URL

Once Tabby is running, you can use the UI with the <u>Direct URL</u> in the **Embed this Space** option (top right).
```

File: docs/hub/spaces-sdks-docker-tabby.md [hunk 3]
```diff
@@ -0,0 +1,60 @@
...
+You’ll see a URL like this: https://tabbyml-tabby.hf.space. This URL gives you access to a full-screen, stable Tabby instance, and is the API Endpoint for IDE / Editor Extensions to talk with.
+
+### Connect VSCode Extension to Space backend
+1. Install the [VSCode Extension](https://marketplace.visualstudio.com/items?itemName=TabbyML.vscode-tabby).
```
Comment (mishig25) [inline docs/hub/spaces-sdks-docker-tabby.md:34]:
```suggestion
### Connect VSCode Extension to Space backend

1. Install the [VSCode Extension](https://marketplace.visualstudio.com/items?itemName=TabbyML.vscode-tabby).
```

**Comment by wsxiaoys** [comment]
> @wsxiaoys could you add your new file in the toctree [here](https://github.com/huggingface/hub-docs/blob/main/docs/hub/_toctree.yml#L151-L173) ?

added

**Comment by wsxiaoys** [comment]
Kindly ping :)

**Comment by wsxiaoys** [comment]
👀

File: docs/hub/spaces-sdks-docker-tabby.md [hunk 4]
```diff
@@ -0,0 +1,65 @@
...
+<Tip>
+
+If you want to customize the title, emojis, and colors of your space, go to "Files and Versions" and edit the metadata of your README.md file.
+
+</Tip>
+
+You’ll see the Building status and once it becomes Running your space is ready to go. If you don’t see the Tabby swagger UI, try refreshing the page.
+
+![image](https://github.com/huggingface/hub-docs/assets/388154/f2320e55-7aad-42ed-8b1c-aec4e42e05f0)
+
```
Comment (osanseviero) [inline docs/hub/spaces-sdks-docker-tabby.md:30]:
```suggestion


You’ll see the *Building status*. Once it becomes *Running*, your Space is ready to go. If you don’t see the Tabby Swagger UI, try refreshing the page.

![image](https://github.com/huggingface/hub-docs/assets/388154/f2320e55-7aad-42ed-8b1c-aec4e42e05f0)

<Tip>

If you want to customize the title, emojis, and colors of your space, go to "Files and Versions" and edit the metadata of your README.md file.

</Tip>

```

File: docs/hub/spaces-sdks-docker-tabby.md [hunk 5]
```diff
@@ -0,0 +1,65 @@
...
+<Tip>
+
+If you want to customize the title, emojis, and colors of your space, go to "Files and Versions" and edit the metadata of your README.md file.
+
+</Tip>
+
+You’ll see the Building status and once it becomes Running your space is ready to go. If you don’t see the Tabby swagger UI, try refreshing the page.
+
+![image](https://github.com/huggingface/hub-docs/assets/388154/f2320e55-7aad-42ed-8b1c-aec4e42e05f0)
```
Comment (osanseviero) [inline docs/hub/spaces-sdks-docker-tabby.md:29]: Please add this asset to https://huggingface.co/datasets/huggingface/documentation-images/tree/main/hub and use it from there

File: docs/hub/spaces-sdks-docker-tabby.md [hunk 6]
```diff
@@ -0,0 +1,65 @@
...
+
+### Your Tabby Space URL
+
+Once Tabby is running, you can use the UI with the <u>Direct URL</u> in the **Embed this Space** option (top right).
```
Comment (osanseviero) [inline docs/hub/spaces-sdks-docker-tabby.md:33]: Note that this only shows if the repo is public

File: docs/hub/spaces-sdks-docker-tabby.md [hunk 7]
```diff
@@ -0,0 +1,65 @@
...
+You’ll see a URL like this: https://tabbyml-tabby.hf.space. This URL gives you access to a full-screen, stable Tabby instance, and is the API Endpoint for IDE / Editor Extensions to talk with.
+
+### Connect VSCode Extension to Space backend
+
+1. Install the [VSCode Extension](https://marketplace.visualstudio.com/items?itemName=TabbyML.vscode-tabby).
+2. Open the file located at `~/.tabby-client/agent/config.toml`. Uncomment both the `[server]` section and the `[server.requestHeaders]` section.
+   * Set the endpoint to the Direct URL you found in the previous step, which should look something like `https://UserName-SpaceName.hf.space`.
+   * As the space is set to **Private**, it is essential to configure the authorization header for accessing the endpoint. You can obtain a token from the [Access Tokens](https://huggingface.co/settings/tokens) page.
```
Comment (osanseviero) [inline docs/hub/spaces-sdks-docker-tabby.md:41]:
```suggestion
   * As the Space is set to **Private**, it is essential to configure the authorization header for accessing the endpoint. You can obtain a token from the [Access Tokens](https://huggingface.co/settings/tokens) page.
```

File: docs/hub/spaces-sdks-docker-tabby.md (hunk 5 above)
Comment (osanseviero) [reply docs/hub/spaces-sdks-docker-tabby.md]: Merged!

**Comment by wsxiaoys** [comment]
Kindly ping @mishig25 ~

**Comment by wsxiaoys** [comment]
Another ping to @mishig25 👀

**Review by mishig25 — APPROVED** [review]
lgtm !

**Comment by mishig25** [comment]
Will merge after CI succeeds https://github.com/huggingface/hub-docs/actions/runs/6238190311/job/17322615240?pr=947

**Comment by mishig25** [comment]
The error is not related to PR. I'm checking it 👁️

**Comment by wsxiaoys** [comment]
```
4:42:15 PM [vite-plugin-svelte] /tmp/tmpx_0uetoq/kit/node_modules/@sveltejs/kit/src/runtime/components/error.svelte:12:8 'IconCopyLink' is not defined
```

Seems triggering an unrelated error - shall i do a rebase?

**Comment by wsxiaoys** [comment]
Rebased against main branch

**Comment by mishig25** [comment]
@wsxiaoys thanks a lot for working on it! The page is live https://huggingface.co/docs/hub/spaces-sdks-docker-tabby
image

**Comment by wsxiaoys** [comment]
Great! Thanks everyone!

**Pull request merged**